lib/improver/tests/utilities/cube_manipulation/test_expand_bounds.py

```diff
@@ -0,0 +1,184 @@
+
+        self.expected_bounds_hours = [
+            date2num(dt(2015, 11, 19, 0), 'hours since 1970-01-01 00:00:00',
+                     CALENDAR),
+            date2num(dt(2015, 11, 19, 3), 'hours since 1970-01-01 00:00:00',
+                     CALENDAR)]
+
+    def test_basic_time_mid(self):
+        """Test that expand_bound produces sensible bounds
+        when given arg 'mid' for times in seconds"""
+        time_point = np.around(date2num(dt(2015, 11, 19, 1, 30), TIME_UNIT,
+                                        CALENDAR)).astype(np.int64)
+        expected_result = iris.coords.DimCoord(
+            [time_point], bounds=self.expected_bounds_seconds,
+            standard_name='time', units=TIME_UNIT)
+        result = expand_bounds(
+            self.cubelist[0], self.cubelist, {'time': 'mid'})
+        self.assertEqual(result.coord('time'), expected_result)
+        self.assertEqual(result.coord('time').dtype, np.int64)
+
+    def test_time_mid_data_precision(self):
+        """Test that expand_bound does not escalate precision when input is
+        of dtype int32"""
+        expected_result = iris.coords.DimCoord(
+            np.array([5400], dtype=np.int32),
+            bounds=np.array([0, 10800], dtype=np.int32),
+            standard_name='forecast_period', units='seconds')
+        result = expand_bounds(
+            self.cubelist[0], self.cubelist, {'forecast_period': 'mid'})
+        self.assertEqual(result.coord('forecast_period'), expected_result)
+        self.assertEqual(result.coord('forecast_period').dtype, np.int32)
+
+    def test_float_time_mid(self):
+        """Test that expand_bound produces sensible bounds
+        when given arg 'mid' for times in hours"""
+        time_unit = 'hours since 1970-01-01 00:00:00'
+        for cube in self.cubelist:
+            cube.coord("time").convert_units(time_unit)
+        time_point = date2num(dt(2015, 11, 19, 1, 30), time_unit, CALENDAR)
+        expected_result = iris.coords.DimCoord(
+            [time_point], bounds=self.expected_bounds_hours,
+            standard_name='time', units=time_unit)
+        result = expand_bounds(
+            self.cubelist[0], self.cubelist, {'time': 'mid'})
+        self.assertEqual(result.coord('time'), expected_result)
+        self.assertEqual(result.coord('time').dtype, np.float32)
+
+    def test_basic_time_upper(self):
+        """Test that expand_bound produces sensible bounds
+        when given arg 'upper'"""
+        time_point = np.around(date2num(dt(2015, 11, 19, 3), TIME_UNIT,
+                                        CALENDAR)).astype(np.int64)
+        expected_result = iris.coords.DimCoord(
+            [time_point], bounds=self.expected_bounds_seconds,
+            standard_name='time', units=TIME_UNIT)
+        result = expand_bounds(
+            self.cubelist[0], self.cubelist, {'time': 'upper'})
+        self.assertEqual(result.coord('time'), expected_result)
+
+    def test_multiple_coordinate_expanded(self):
```

Comment:
This is the only new test I have added, the rest are essentially copy-paste (except modified to take a dictionary).